lib/rust/ensogl/component/drop-down/src/entry.rs

```diff
@@ -0,0 +1,204 @@
+        Self { display_object, label_thin, label_bold }
+    }
+
+    fn update_selected(&self, selected: bool) {
+        if selected {
+            self.display_object.remove_child(&self.label_thin);
+            self.display_object.add_child(&self.label_bold);
+        } else {
+            self.display_object.remove_child(&self.label_bold);
+            self.display_object.add_child(&self.label_thin);
+        }
+    }
+
+    fn update_layout(&self, contour: entry::Contour, text_size: text::Size, text_offset: f32) {
+        let label_pos = Vector2(text_offset - contour.size.x / 2.0, text_size.value / 2.0);
+        self.label_thin.set_xy(label_pos);
+        self.label_bold.set_xy(label_pos);
+    }
+}
+
+
+// === Entry ===
+
+/// A [`SimpleGridView`] entry - a label with background.
+#[derive(Clone, CloneRef, Debug)]
+pub struct Entry {
+    frp:  EntryFrp<Self>,
+    data: Rc<EntryData>,
+}
+
+impl ensogl_grid_view::Entry for Entry {
+    type Model = EntryModel;
+    type Params = EntryParams;
+
+    fn new(app: &Application, text_layer: Option<&Layer>) -> Self {
+        let data = Rc::new(EntryData::new(app, text_layer));
+        let frp = EntryFrp::<Self>::new();
+        let input = &frp.private().input;
+        let out = &frp.private().output;
+        let network = frp.network();
+
+        out.hover_highlight_color.emit(color::Lcha::transparent());
+
+        enso_frp::extend! { network
+            size <- input.set_size.on_change();
+            font <- input.set_params.map(|p| p.font.clone_ref()).on_change();
+            text_offset <- input.set_params.map(|p| p.text_offset).on_change();
+            focus_color <- input.set_params.map(|p| p.focus_color).on_change();
+            text_color <- input.set_params.map(|p| p.text_color).on_change();
+            text_size <- input.set_params.map(|p| p.text_size).on_change();
+            corners_radius <- input.set_params.map(|p| p.corners_radius).on_change();
+            selected_text_color <- input.set_params.map(|p| p.selected_text_color).on_change();
+
+            contour <- all_with(&size, &corners_radius, |&size, &corners_radius| entry::Contour { size, corners_radius });
+            layout <- all(contour, text_size, text_offset);
+            eval layout ((&(c, ts, to)) data.update_layout(c, ts, to));
+            selected <- input.set_model.map(|m| *m.selected);
+            eval selected ((&s) data.update_selected(s));
+
+            text_size <- text_size.ref_into_some();
+            data.label_thin.set_property_default <+ text_size;
+            data.label_bold.set_property_default <+ text_size;
+            data.label_thin.set_property_default <+ text_color.ref_into_some();
+            data.label_bold.set_property_default <+ selected_text_color.ref_into_some();
+            data.label_thin.set_font <+ font;
+            data.label_bold.set_font <+ font;
+
+            natural_entry_width <- data.label_bold.width.map2(&text_offset, |w, offset| w + offset);
```

Comment:
what is "natural" width?

Reply:
Natural here refers to the width that the entry would like to take by itself, if no external limits were in place. It is based on text label width, with paddings applied.

I'm not sure what could be a more descriptive name, without being too verbose. Maybe `desired_entry_width`, but I'm not sure if that's better/easier to understand.